File: Java/Android Architecture Component/DataBinding-Tabian Gifts/app/src/main/java/com/contextidea/tabiangifts/MainActivity.java
package com.contextidea.tabiangifts;


import android.databinding.DataBindingUtil;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.contextidea.tabiangifts.databinding.ActivityMainBinding;


public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    //data binding
    ActivityMainBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        init();
    }

    private void init() {
        ViewProductFragment fragment=new ViewProductFragment();
        FragmentTransaction transaction=getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.main_container,fragment,"Main");
        transaction.commit();
    }

}

File: Kotlin/Kotlin Retrofit2 RxJava/app/src/main/java/com/example/zaynannisa/kotlinretrofitrxjava/MainActivity.kt
package com.example.zaynannisa.kotlinretrofitrxjava

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager

import com.example.zaynannisa.kotlinretrofitrxjava.adapter.PostAdapter
import com.example.zaynannisa.kotlinretrofitrxjava.data.Post
import com.example.zaynannisa.kotlinretrofitrxjava.retrofit.IMyApi
import com.example.zaynannisa.kotlinretrofitrxjava.retrofit.RetrofitClient

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
    internal lateinit var jsonApi:IMyApi
    internal lateinit var compositeDisposable: CompositeDisposable
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        compositeDisposable=CompositeDisposable()
        //init api
        val retrofit=RetrofitClient.instance
        jsonApi=retrofit.create(IMyApi::class.java)
        //Recycler View
        recyler_post.setHasFixedSize(true)
        recyler_post.layoutManager=LinearLayoutManager(this )
        fetchData()
    }

    private fun fetchData() {
        compositeDisposable.add(jsonApi.posts
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe{posts->displayData(posts)}
        )

    }

    private fun displayData(posts: List<Post>?) {
     val adapter=PostAdapter(this,posts!!)
        recyler_post.adapter=adapter
    }
}
